pvlib/pvarray.py

```diff
@@ -348,3 +348,68 @@ def huld(effective_irradiance, temp_mod, pdc0, k=None, cell_type=None):
                     k[2] * tprime + k[3] * tprime * logGprime +
                     k[4] * tprime * logGprime**2 + k[5] * tprime**2)
     return pdc
+
+
+def _build_iec61853():
+    ee = np.array([100, 100, 200, 200, 400, 400, 400, 600, 600, 600, 600,
+                   800, 800, 800, 800, 1000, 1000, 1000, 1000, 1100, 1100,
+                   1100]).T
+    tc = np.array([15, 25, 15, 25, 15, 25, 50, 15, 25, 50, 75,
+                   15, 25, 50, 75, 15, 25, 50, 75, 25, 50, 75]).T
+    return ee, tc
+
+
+def fit_huld(effective_irradiance, temp_mod, pdc):
+    r'''
+    Fit the Huld model to the input data.
+
+    Parameters
+    ----------
+    effective_irradiance : numeric
+        The irradiance that is converted to photocurrent. [:math:`W/m^2`]
+    temp_mod: numeric
+        Module back-surface temperature. [C]
+    pdc: numeric
+        DC power at ``effectuve_irradiance`` and ``temp_mod``. [W]
+
+    Returns
+    -------
+    pdc0: numeric
+        Power of the modules at reference conditions 1000 :math:`W/m^2`
+        and :math:`25^{\circ}C`. [W]
+    k : tuple
+        Empirical coefficients used in the Huld model. Length 6.
+
+    Notes
+    -----
+    Requires ``statsmodels``.
+
+    See Also
+    --------
+    pvlib.pvarray.huld
```

Comment:
Should we have a `References` entry to the Huld paper in this docstring too?

Reply:
AFAIK Huld does not fully describe his fitting method, so a discussion may be needed to assess whether this requires a new publication. See #1898.

Reply:
Here's what's in Huld's 2011 paper:

"The fit is done by a least-square procedure. Although such a procedure exists in many software packages we have made use of the fact that Eq.(1) is linear in the coefficients, and developed a very fast algorithm dedicated to performing the least-square fit. The least-square fit results in a set of 7 linear equations that has a unique solution. The description of the algorithm is not central to this investigation, but may be consulted on the webpage with supplementary material for this paper: http://re.jrc.ec.europa.eu/esti/powermo del/index.html. The source code for the algorithm may also be downloaded there, together with the measurement data used for the fitting of the generic model."

From that description I suspect the method is OLS. But the URL is extinct and waybackmachine doesn't have it in the archive. I've reached out to authors to clarify.

Reply:
I tried to find out more when I wrote about this model.  Hopefully you will have more success!

Reply:
Sadly, I could get no further information.

Reply:
I have added a kwarg to select OLS or a robust method, and a note explaining that the method in the reference isn't identified, so pvlib is providing these options.